core/src/main/scala/cilib/Entity.scala

```diff
@@ -21,79 +21,87 @@ object Entity {
     }
 }
 
-object BCHM {    
- def clamp[S,A:scalaz.Equal](x: Entity[S,A])(implicit N: spire.math.Numeric[A]) =
+object BCHM {
+  def clamp[S, A: scalaz.Equal](x: Entity[S, A])(implicit N: spire.math.Numeric[A]) =
     Lenses._position.modify((p: Position[A]) => {
-      val c: NonEmptyList[A] = p.pos zip p.boundary map {
+      val c: NonEmptyList[A] = p.pos.zip(p.boundary).map {
         case t if N.toDouble(t._1) < t._2.lowerValue => N.fromDouble(t._2.lowerValue)
         case t if N.toDouble(t._1) > t._2.upperValue => N.fromDouble(t._2.upperValue)
-        case t if t._2.contains(N.toDouble(t._1)) => t._1
+        case t if t._2.contains(N.toDouble(t._1))    => t._1
```

Comment:
So, what would be the behavior if `t._1` is not contained in `t._2`? However unlikely that _should_ be? I'm curious as this is a partial function, and they usually bite you later.

Reply:
Hmm I see what you're saying. I'll put the first two cases, the greater than and less than cases, in to a parent case that is a doesNotContain. If that makes sense?